### PR TITLE
Fix segmentation fault if crypt() returns error.

### PR DESCRIPTION
**Pull request by jannic**
This issue has been reported to the debian bug tracker at
https://bugs.debian.org/863650

### CODE REVIEW

File: src/backend_pgsql.c [hunk 1]
```diff
@@ -270,7 +270,7 @@ backend_authenticate(const char *service, const char *user, const char *passwd,
 						}
 					} else {
 						tmp = password_encrypt(options, user, passwd, stored_pw);
-						if (!strcmp(stored_pw, tmp)) {
+						if (tmp != NULL && !strcmp(stored_pw, tmp)) {
 							rc = PAM_SUCCESS;
 						}
 						free (tmp);
```
Comment (william-gr) [inline src/backend_pgsql.c:276]: You're still trying to free tmp

**Review by william-gr — CHANGES_REQUESTED** [review]
Thanks, just a minor note

File: src/backend_pgsql.c (hunk 1 above)
Comment (jannic) [reply src/backend_pgsql.c:276]: Right, but that shouldn't matter: free() is explicitly allowed to be called with NULL as a parameter. From the manpage: "If ptr is NULL, no operation is performed."
Would you prefer if I changed to code this way?
```
                                                tmp = password_encrypt(options, user, passwd, stored_pw);
                                                if (tmp != NULL ) {
                                                        if (!strcmp(stored_pw, tmp)) {
                                                                rc = PAM_SUCCESS;
                                                        }
                                                        free (tmp);
                                                }
```

**Review by william-gr — APPROVED** [review]
Well, its one less call, but nevermind. Thanks for your contribution

**Comment by errge** [comment]
@william-gr Could you please cut a release containing this fix?

This would be super useful for the community, because currently this package is blacklisted from hitting any Debian Releases ever again, as https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863650 is still open and without a new upstream release it won't be closed.  Since the severity is grave, the package never moves to testing: https://tracker.debian.org/pkg/pam-pgsql

I think the only way out of this situation is to cut a release, so we have a proper version number and the Debian maintainer can then be pinged to add the release to debian and close the bug, after which the package should be automatically included in the next stable release again (after missing both buster and bullseye).

Thank you for your time and work on this project!

**Comment by william-gr** [comment]
Unfortunately I dont have spare time to work on this currently. I will be happy to provide access for a maintainer of debian to do it.

**Pull request merged**